triton-vm/src/table/processor_table.rs

```diff
@@ -2425,6 +2556,188 @@ impl RowPairConstraints {
+            - self.running_evaluation_standard_output();
+
+        write_io_selector * running_evaluation_remains
+            + write_io_deselector * running_evaluation_updates
+    }
+
+    pub fn running_product_for_op_stack_table_updates_correctly(
+        &self,
+        challenges: &ProcessorTableChallenges,
+    ) -> MPolynomial<XFieldElement> {
+        let indeterminate = self.constant_x(challenges.op_stack_perm_indeterminate);
+        let clk_weight = self.constant_x(challenges.op_stack_table_clk_weight);
+        let ib1_weight = self.constant_x(challenges.op_stack_table_ib1_weight);
+        let osp_weight = self.constant_x(challenges.op_stack_table_osp_weight);
+        let osv_weight = self.constant_x(challenges.op_stack_table_osv_weight);
+        let compressed_row = clk_weight * self.clk_next()
+            + ib1_weight * self.ib1_next()
+            + osp_weight * self.osp_next()
+            + osv_weight * self.osv_next();
+
+        self.running_product_op_stack_table_next()
+            - self.running_product_op_stack_table() * (indeterminate - compressed_row)
+    }
+
+    pub fn running_product_for_ram_table_updates_correctly(
+        &self,
+        challenges: &ProcessorTableChallenges,
+    ) -> MPolynomial<XFieldElement> {
+        let indeterminate = self.constant_x(challenges.ram_perm_indeterminate);
+        let clk_weight = self.constant_x(challenges.ram_table_clk_weight);
+        let ramp_weight = self.constant_x(challenges.ram_table_ramp_weight);
+        let ramv_weight = self.constant_x(challenges.ram_table_ramv_weight);
+        let compressed_row = clk_weight * self.clk_next()
+            + ramp_weight * self.ramp_next()
+            + ramv_weight * self.ramv_next();
+
+        self.running_product_ram_table_next()
+            - self.running_product_ram_table() * (indeterminate - compressed_row)
+    }
+
+    pub fn running_product_for_jump_stack_table_updates_correctly(
+        &self,
+        challenges: &ProcessorTableChallenges,
+    ) -> MPolynomial<XFieldElement> {
+        let indeterminate = self.constant_x(challenges.jump_stack_perm_indeterminate);
+        let clk_weight = self.constant_x(challenges.jump_stack_table_clk_weight);
+        let ci_weight = self.constant_x(challenges.jump_stack_table_ci_weight);
+        let jsp_weight = self.constant_x(challenges.jump_stack_table_jsp_weight);
+        let jso_weight = self.constant_x(challenges.jump_stack_table_jso_weight);
+        let jsd_weight = self.constant_x(challenges.jump_stack_table_jsd_weight);
+        let compressed_row = clk_weight * self.clk_next()
+            + ci_weight * self.ci_next()
+            + jsp_weight * self.jsp_next()
+            + jso_weight * self.jso_next()
+            + jsd_weight * self.jsd_next();
+
+        self.running_product_jump_stack_table_next()
+            - self.running_product_jump_stack_table() * (indeterminate - compressed_row)
+    }
+
```

Comment:
With the picture you drew on the whiteboard, it became clear that this really should be `instruction_deselector_next` here and without `_next` in the next function. Consider adding the picture to the spec.